### ISSUE TITLE
Accessing addEvent from host

### ISSUE THREAD

**Issue by myreauks** [cornedriesprong/AUv3SequencerExample#1]
Hi! 
Thank you so much for the great example code. 

This is not an issue per se, but I would greatly appreciate if you have pointers on how to access the addEvent and deleteEvent functions from the host side?, except from the viewcontroller of the auv3 itself?

Is this even possible, or does the AUAudioUnit restrict access inherently? I've been banging my head to the wall with this issue for a couple of days as I'm mostly used to Swift development and Objective-C seems really tricky to me.

On a sidenote, why is the function called as just add() in the viewController even though it is defined as addNote()?

Thanks for any help you can provide!

**Comment by cornedriesprong** [comment]
You can just call them as methods on an instance of your AUAudioUnit subclass, since Swift and Objective-C can interoperate. The `addEvent` Objective-C method gets translated to `add` because of different conventions in both languages I suppose.

See [this](https://github.com/cornedriesprong/AUv3SequencerExample/blob/24a5f40d15f9c69d2bc077ac79912153e6cdbd26/AUv3SequencerExampleAudioUnit/UI/AudioUnitViewController.swift#L22) example code. If the methods that you defined on your audio unit subclass are not visible from Swift, make sure they are defined in the header of the Objective-C class (and therefore public) and that you cast the generic `AUAudioUnit` to your custom subclass.

As a sidenote, you might just as well write your audio unit in Swift, you just have make sure that the code in your `internalRenderBlock` callback is realtime-safe (and so it's safest to write it in C, which is easier to interoperate with from Objective-C), see [this article](http://atastypixel.com/four-common-mistakes-in-audio-development/) for more details.

**Comment by myreauks** [comment]
Thank you for the fast reply and the help!

The typecasting is the part that I was missing. I tried to get it working with AVAudioUnit that I instantiate with the AUv3SequencerExampleAudioUnit. Using the AVAudioUnits auAudioUnit fails with the typecasting. Maybe I'm doing this wrong or is it not possible to do it this way.

Anyhow, I found an alternate way to move extra data to the AUv3 by exploiting the MIDI UMP protocol so this is not an absolute must for me to get working this way. Might be useful in the future though. I'll close this as solved for now.

Thanks for the help!

Best,
Miro
